lib-multisrc/kemono/src/eu/kanade/tachiyomi/multisrc/kemono/Kemono.kt

```diff
@@ -63,69 +64,120 @@ open class Kemono(
+        }
+
         val response = client.newCall(GET("$baseUrl/$apiPath/creators", headers)).execute()
-        val allCreators = block(response.parseAs())
-        val count = allCreators.size
-        val fromIndex = (page - 1) * NEW_PAGE_SIZE
-        val toIndex = min(count, fromIndex + NEW_PAGE_SIZE)
-        val creators = allCreators.subList(fromIndex, toIndex)
-            .map { it.toSManga(imgCdnUrl) }
-            .filterUnsupported()
-        return MangasPage(creators, toIndex < count)
+        val allCreators = response.parseAs<List<KemonoCreatorDto>>().filterNot { it.service.lowercase() == "discord" }
+        val mangas = allCreators.filter {
+            val includeType = typeIncluded.isEmpty() || typeIncluded.contains(it.service.serviceName().lowercase())
+            val excludeType = typeExcluded.isNotEmpty() && typeExcluded.contains(it.service.serviceName().lowercase())
+
+            val regularSearch = it.name.contains(title, true)
+
+            val isFavourited = when (fav) {
+                true -> favourites.any { f -> f.id == it.id.also { _ -> it.fav = f.faved_seq } }
+                false -> favourites.none { f -> f.id == it.id }
+                else -> true
+            }
+
+            includeType && !excludeType && isFavourited &&
+                regularSearch
+        }
+
+        val sorted = when (sort.first) {
+            "pop" -> {
+                if (sort.second == "desc") {
+                    mangas.sortedByDescending { it.favorited }
+                } else {
+                    mangas.sortedBy { it.favorited }
+                }
+            }
+            "tit" -> {
+                if (sort.second == "desc") {
+                    mangas.sortedByDescending { it.name }
+                } else {
+                    mangas.sortedBy { it.name }
+                }
+            }
+            "new" -> {
+                if (sort.second == "desc") {
+                    mangas.sortedByDescending { it.id }
+                } else {
+                    mangas.sortedBy { it.id }
+                }
+            }
+            "fav" -> {
+                if (fav != true) throw Exception("Please check 'Favourites Only' Filter")
+                if (sort.second == "desc") {
+                    mangas.sortedByDescending { it.fav }
+                } else {
+                    mangas.sortedBy { it.fav }
+                }
+            }
+            else -> {
+                if (sort.second == "desc") {
+                    mangas.sortedByDescending { it.updatedDate }
+                } else {
+                    mangas.sortedBy { it.updatedDate }
+                }
+            }
+        }
+        val maxIndex = mangas.size
+        val fromIndex = (page - 1) * PAGE_CREATORS_LIMIT
+        val toIndex = min(maxIndex, fromIndex + PAGE_CREATORS_LIMIT)
+
+        val final = sorted.subList(fromIndex, toIndex).map { it.toSManga(imgCdnUrl) }
+        return MangasPage(final, toIndex != maxIndex)
```

Comment:
cache the results in the class to avoid fetching again on next page

Reply:
Hello, thanks for the suggestion. But can explain to me on how I could achieve that? Thank you :pray:

Reply:
basically cache result in variable

https://github.com/keiyoushi/extensions-source/blob/00bab8e7764de72d91c764688666535ffaafa051/src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/Comick.kt#L231-L236

Reply:
Thank you :pray: